sdk/storage/inc/datalake/datalake_utilities.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/crypt.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  inline std::map<std::string, std::string> ParseDataLakePropertiesString(
+      const std::string& dataLakePropertiesString)
+  {
+    std::map<std::string, std::string> result;
+
+    auto startItr = dataLakePropertiesString.begin();
+    auto endItr = startItr + dataLakePropertiesString.find(",");
```

Comment:
`startItr` is an `std::iterator` but `endItr` is not, right? But you're using the same name pattern here, which is kind of confusing.

Besides, is `Itr` a common short form for `iterator`?

Reply:
What if there's only one k-v pair in properties? There would be no ','.

Reply:
I suggest you use `std::find()` here.

Reply:
I think it will work properly, if there is no `,`, since `dataLakePropertiesString.end()` will be endItr.